### DISCUSSION TITLE
F435 in F415 PCB

### DISCUSSION THREAD

**Discussion by CDRXavier** [keirf/flashfloppy#918] [Q&A] [answered]
Hello I am wondering if it is possible to change the controller in the F415 models to the F435. Models with the F435 seems impossibly hard to find. It's mostly F415s and maybe F105s.

Usually for MCUs with pure performance upgrades it is pin compatible and is the same package, but just making sure.

Do you support directly flashing to a blank MCU? Given that we are using the like, official IDE and flashing tools, I imagine it should be able to hardware program via SPI.

The flashing part is important, because if I buy F435s from supplier, they will come blank, and they may not ship with USB bootloader flashed.

I know like the entire soldering and stuff, I have access to hot air and even reflow oven.

**Comment by CDRXavier** [comment]
I noticed there are PCB revisions with different footprints for the chips. I don't know if I will be getting one with the QFP64.

I can find models with PCBs marked 922D, KC30AT2 and KC30AT3. I imagine I would prefer the 922D pcb for the footprint reasons?

**Comment by keirf** [comment, marked as answer]
KC30AT2 = AT32F415 QFN32
KC30AT3 = AT32F415 LQFP48
KC30AT4 = AT32F415 LQFP64
922D = STM32F105 LQFP64

Actually only the last of these is compatible with AT32F435. The 435 cannot be used in a 415 design because it has more power pins. However 435 (and 415) can be used in a 105 design.

**Reply by CDRXavier** [reply]
Aah. so I should get a 922D board. Cool.

**Comment by keirf** [comment]
However you can't rely on seller photos. You'll get whatever they've got.
